drivers/windows/file_access_windows.cpp

```diff
@@ -234,6 +236,12 @@ bool FileAccessWindows::eof_reached() const {
 uint8_t FileAccessWindows::get_8() const {
 
 	ERR_FAIL_COND_V(!f, 0);
+	if (flags > 2) { // READ_WRITE or WRITE_READ
```

Comment:
That's a bit dangerous if we decide to renumber/reorder those flags.
Would be better to check explicitly for `READ_WRITE` and `WRITE_READ` in the bitmask.

Reply:
Fixed !

Reply:
Sorry I was not explicit enough, I meant to do bitwise checks with `(flags & READ_WRITE) || (flags & WRITE_READ)`.

Reply:
They are not really formatted as flags, in fact its just an enum

Reply:
I think they're meant to be formatted as flags, see:
```
        enum ModeFlags {

                READ = 1,
                WRITE = 2,
                READ_WRITE = 3,
                WRITE_READ = 7,
        };
```
```
READ_WRITE = READ | WRITE
WRITE_READ = 4 | READ_WRITE
```
(so if you use `READ | WRITE` in your code, it should behave the same as `READ_WRITE` which is just sugar for it. And `WRITE_READ` includes `READ_WRITE` too, but is more specialized via the `4` bit)

Reply:
Hmm.. ok, I can of course change it as you said, however the flags are a bit different and should have power of 2 for example(in C#):

> [Flags]
> public enum Options 
> {
>     None    = 0,
>     Option1 = 1,
>     Option2 = 2,
>     Option3 = 4,
>     Option4 = 8
> }
>

Reply:
But that's the case here, `READ` and `WRITE` are powers of 2, and `READ_WRITE` is just `READ | WRITE` (so it's a combination of flags).
And `WRITE_READ` is also a combination of powers of 2, namely `4 | READ | WRITE`. `4` is not used by itself, but it's used to make `WRITE_READ` a superset of `READ_WRITE`.

Reply:
But well, those flags are always checked with `==` so far so I guess it's fine as is.

Reply:
Ah, you right :S there are two ways to solve same problem..